### PR TITLE
Fix OOM issue in AIShell recipe 

### PR DESCRIPTION
**Pull request by danpovey**

### CODE REVIEW

**Comment by danpovey** [comment]
Relates to #114 and #75

**Comment by danpovey** [comment]
This (reduction of beam from 10 to 6 in AIShell recipe) together with some code changes I just committed in k2, seems to resolve the OOM issue.

@csukuangfj there are some changes in mmi_graph.py which are from you.   Would you mind verifying that this is OK to merge?

**Comment by csukuangfj** [comment]
> @csukuangfj there are some changes in mmi_graph.py which are from you. Would you mind verifying that this is OK to merge?


Sure, will run MMI training for AIShell with this pullrequest.

**Comment by danpovey** [comment]
@csukuangfj  I mainly meant looking at the code or letting me know that this was intended for merge.
I am running the  AIShell recipe.
(I need to check that librispeech does not crash though-- no GPUs left).

**Comment by csukuangfj** [comment]
OK, looking into it. I tested it before when I made the changes and the `mmi_bigram_train.py` got a similar loss as 
the script before the changes.

File: egs/librispeech/asr/simple_v1/mmi_att_transformer_train.py [hunk 1]
```diff
@@ -424,14 +424,19 @@ def main():
             L_inv = k2.arc_sort(L.invert_())
             torch.save(L_inv.as_dict(), lang_dir / 'Linv.pt')
 
+    device_id = 0
```
Comment (csukuangfj) [inline egs/librispeech/asr/simple_v1/mmi_att_transformer_train.py:427]: Line 274
```python
assert P.is_cpu
```
has to be deleted.

**Comment by csukuangfj** [comment]
The changes look good to me.

**Comment by danpovey** [comment]
Added AIShell results.
Note, the CER  is 6.94%, while the best Kaldi number I see in egs/aishell/s5 is around 7.4%.  So if I am not mistaken we may have passed Kaldi's performance on this particular dataset.

**Comment by danpovey** [comment]
And many thanks to @heguocai for his contribution!

**Pull request merged**